src/python/pants/base/fingerprint_strategy.py

```diff
@@ -39,6 +39,8 @@ def compute_fingerprint(self, target):
   def fingerprint_target(self, target):
     """Consumers of subclass instances call this to get a fingerprint labeled with the name"""
     fingerprint = self.compute_fingerprint(target)
+    # TODO: We probably don't need to qualify the fingerprint with the strategy name any more,
```

Comment:
Hmm, now that I think of it further, is my change actually OK?  I'm thinking of this scenario:

1) Build with fprint strategy A.
2) Make some changes, and build them with fprint strategy B. Everything gets rebuilt.
3) Revert the changes, and rebuild with fprint strategy A. Nothing gets rebuilt, because the invalidator key file from 1) is still where we left it. So now Pants thinks the workdir is back to state 1), but it's actually in state 2).

I may have neglected to consider the fact that overwriting the existing key is part of the behavior.

Any alternative ideas on how to solve this?